packages/server/src/questionType/questionType.controller.ts

```diff
@@ -63,20 +64,43 @@ export class QuestionTypeController {
     });
     if (!questionType) {
       try {
-        await QuestionTypeModel.create({
-          cid: courseId,
-          name: newQuestionType.name,
-          color: newQuestionType.color,
-          queueId: queueId,
-        }).save();
-        res.status(HttpStatus.OK).send('success');
+        await getManager().transaction(async (transactionalEntityManager) => {
+          await transactionalEntityManager
+            .create(QuestionTypeModel, {
+              cid: courseId,
+              name: newQuestionType.name,
+              color: newQuestionType.color,
+              queueId: queueId,
+            })
+            .save();
+          if (queueId) {
+            // update the queue's config to include the new question type
+            const queue = await transactionalEntityManager.findOne(
+              QueueModel,
+              queueId,
+            );
+            queue.config = queue.config || {}; // just in case it's null
+            queue.config.tags = queue.config.tags || {}; // just in case it's undefined
+
+            const nameNoSpecialChars = newQuestionType.name.replace(
```

Comment:
adjusted this to only just escape/remove illegal json characters. Good catch with this btw, I realised this also allowed for empty tag names or duplicates (which wouldn't actually be a duplicate, it would just replace the other tag in the config, which would unsync the question types and config). This is all fixed now